validator/sawtooth_validator/journal/publisher.py

```diff
@@ -379,6 +383,11 @@ def __init__(self,
         self._permission_verifier = permission_verifier
         self._batch_injector_factory = batch_injector_factory
 
+        # For metric gathering
+        self._pending_batch_gauge = \
+            metrics_registry.gauge('pending_batch_gauge') \
+            if metrics_registry else None
```

Comment:
This looks like it will most likely fail later on when there's no metrics registry.  Is there a NO OP metrics registry that can be used when we don't have the OpenSTDB settings in place?

Reply:
Fixed add method `_set_gauge` to `BlockPublisher`